src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -378,12 +446,13 @@ BookStep<TIn, TOut>::revImp (
       Return true to continue to receive offers, false to stop receiving offers.
     */
     auto eachOffer =
-        [&](TOffer<TIn, TOut>& offer,
-            TAmounts<TIn, TOut> const& ofrAmt,
-            TAmounts<TIn, TOut> const& stpAmt,
-            TOut const& ownerGives,
-            std::uint32_t transferRateIn,
-            std::uint32_t transferRateOut) mutable -> bool
+        [this, &sb, &out, &result, &remainingOut, &savedIns, &savedOuts]
```

Comment:
I can certainly go your way on this; the code will execute the same.

Personally, I like Scott Meyer's advice to "Avoid default capture modes" (Item 31).  In part, when I was modifying `eachOffer`, it was illuminating to see just how much it reaches outside its own scope.  This list shows me explicitly what `eachOffer` is messing with.  I liked that, but I won't insist on it.